AppController/djinn.rb

```diff
@@ -2001,23 +2001,30 @@ def job_start(secret)
     # This pid is used to control this deployment using the init script.
     HelperFunctions.write_file(PID_FILE, "#{Process.pid}")
 
+    # We reload our old IPs (if we find them) so we can check later if
+    # they changed and act accordingly.
+    if File.exists?("#{APPSCALE_CONFIG_DIR}/my_private_ip")
+      @my_private_ip = HelperFunctions.read_file("#{APPSCALE_CONFIG_DIR}/my_private_ip").chomp
+    end
+    if File.exists?("#{APPSCALE_CONFIG_DIR}/my_public_ip")
+      @my_public_ip = HelperFunctions.read_file("#{APPSCALE_CONFIG_DIR}/my_public_ip").chomp
+    end
```

Comment:
If the file is removed after the check and before the read, it will raise an exception. I think it's safer to just read it and rescue `Errno::ENOENT`.